### PR TITLE
change read/write functions to use std::byte

### PR DESCRIPTION
**Pull request by neheb**
Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

File: src/iohandler/file_io_handler.cc [hunk 1]
```diff
@@ -46,7 +46,7 @@ void FileIOHandler::open(enum UpnpOpenFileMode mode)
     f = file.open("rb");
 }
 
-std::size_t FileIOHandler::read(char* buf, std::size_t length)
+std::size_t FileIOHandler::read(std::byte* buf, std::size_t length)
 {
     std::size_t ret = std::fread(buf, sizeof(char), length, f);
```
Comment (KarlStraussberger) [inline src/iohandler/file_io_handler.cc:51]: shouldn't this be `std::byte` as well

File: src/util/jpeg_resolution.cc [hunk 2]
```diff
@@ -69,7 +69,7 @@ static int get16m(const std::byte* shrt)
 static int iohFgetc(IOHandler& ioh)
 {
     std::byte c[1] {};
-    int ret = ioh.read(reinterpret_cast<char*>(c), sizeof(char));
+    int ret = ioh.read(c, sizeof(char));
```
Comment (KarlStraussberger) [inline src/util/jpeg_resolution.cc:72]: `std::byte` ?

File: src/iohandler/file_io_handler.cc [hunk 3]
```diff
@@ -60,7 +60,7 @@ std::size_t FileIOHandler::read(char* buf, std::size_t length)
     return ret;
 }
 
-std::size_t FileIOHandler::write(char* buf, std::size_t length)
+std::size_t FileIOHandler::write(std::byte* buf, std::size_t length)
 {
     return std::fwrite(buf, sizeof(char), length, f);
```
Comment (KarlStraussberger) [inline src/iohandler/file_io_handler.cc:65]: `std::byte` ?

**Review by KarlStraussberger — COMMENTED** [review]
Please check out for `sizeof(char)`

File: src/iohandler/file_io_handler.cc (hunk 1 above)
Comment (neheb) [reply src/iohandler/file_io_handler.cc:51]: Good catch.

**Comment by neheb** [comment]
Fixed.

**Pull request merged**